docs/scripts/gen_ref_pages.py

```diff
@@ -0,0 +1,43 @@
+"""Generate the code reference pages and navigation."""
```

Comment:
This could obviously do with a little more documentation itself.  It would be really nice to also pull this out into something that is easier to re-use elsewhere.  E.g. a function that takes in the root path and the module name?